File: src/detecprotecode.java
//
// Copyright (c) Microsoft. All rights reserved.
// To learn more, please visit the documentation - Quickstart: Azure Content Safety: https://aka.ms/acsstudiodoc
//

package com.microsoft.cognitiveservices;
import okhttp3.*;

public class ContentSafetyDetectProtectedMaterialForCodeSampleCode {

    public static void main(String[] args) throws Exception {
        // Replace with your own subscription_key and endpoint
        String subscriptionKey = "your_subscription_key";
        String endpoint = "https://your_resource_name.cognitiveservices.azure.com";
        String aadToken = "your_aad_token";

        // The code to be analyzed
        String codeToAnalyze = "test-code";

        // Set up the API request
        String apiUrl = endpoint + "/contentsafety/text:detectProtectedMaterialForCode?api-version=2024-09-15-preview";

        String requestBody = "{\"code\": \"" + codeToAnalyze + "\"}";

        // Build request
        Headers.Builder headersBuilder = new Headers.Builder();
        //Choose to use key authorization or AAD token authorization
        headersBuilder.add("Ocp-Apim-Subscription-Key", subscriptionKey);
        headersBuilder.add("Authorization", aadToken);
        Request request = new Request.Builder()
                .url(apiUrl)
                .headers(headersBuilder.build())
                .post(RequestBody.create(requestBody, okhttp3.MediaType.parse("application/json; charset=utf-8")))
                .build();

        OkHttpClient client = new OkHttpClient();

        // Send request and get result
        try (Response response = client.newCall(request).execute()) {
            if (response.body() == null) {
                throw new Exception( "Response body is null.");
            }
            String responseText = response.body().string();
            System.out.println(responseText);

            if (!response.isSuccessful()) {
                    throw new Exception(String.valueOf(response.code()));
            }
        }
    }
}

File: src/detecprotecode.py
#
# Copyright (c) Microsoft. All rights reserved.
# To learn more, please visit the documentation - Quickstart: Azure Content Safety: https://aka.ms/acsstudiodoc
#

import requests

# Replace with your own subscription_key/aad_token and endpoint
# Choose to use key authorization or AAD token authorization
subscription_key = "your_subscription_key"
endpoint = "your_resource_endpoint"
aad_token = "your_aad_token"


# The code to be analyzed
code_to_analyze = """your_code_here"""

# Set up the API request
url = f"{endpoint}/contentsafety/text:detectProtectedMaterialForCode?api-version=2024-09-15-preview"        
headers = {
  "Content-Type": "application/json",
  "Ocp-Apim-Subscription-Key": subscription_key,
  "Authorization": aad_token
}
data = {
  "code": code_to_analyze
}

# Send the API request
response = requests.post(url, headers=headers, json=data)

# Handle the API response
if response.status_code == 200:
    result = response.json()
    print("Analysis result:", result)
else:
    print("Error:", response.status_code, response.text)